html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\reliefweb_import\Service;
+
+use Drupal\reliefweb_entities\Entity\Job;
+use Drupal\reliefweb_import\Exception\ReliefwebImportException;
+use Drupal\reliefweb_import\Exception\ReliefwebImportExceptionSoftViolation;
+use Drupal\reliefweb_import\Exception\ReliefwebImportExceptionViolation;
+
+/**
+ * Service to interact with the WorkDay API.
+ */
+class WorkDayJobImporter extends JobFeedsImporterBase implements JobFeedsImporterInterface {
+
+  /**
+   * The settings for the WorkDay service.
+   *
+   * @var array
+   */
+  protected array $settings;
+
+  /**
+   * Set the settings for the WorkDay service.
+   *
+   * @param array $settings
+   *   The settings to use for the WorkDay service.
+   */
+  public function setSettings(array $settings): void {
+    $this->settings = $settings;
+  }
+
+  /**
+   * Get authorization token from WorkDay.
+   */
+  public function getAuthToken(): string {
+    $timeout = $this->settings['timeout'] ?? 10;
+    $base_url = $this->settings['base_url'] ?? '';
+    $tenant = $this->settings['tenant'] ?? '';
+    $client_id = $this->settings['client_id'] ?? '';
+    $client_secret = $this->settings['client_secret'] ?? '';
+    $refresh_token = $this->settings['refresh_token'] ?? '';
+    $token_url = $base_url . '/ccx/oauth2/' . $tenant . '/token';
+
+    $response = $this->httpClient->request('POST', $token_url, [
+      'connect_timeout' => $timeout,
+      'timeout' => $timeout,
+      'headers' => [
+        'Content-Type' => 'application/x-www-form-urlencoded',
+        'Accept' => 'application/json',
+      ],
+      'form_params' => [
+        'grant_type' => 'refresh_token',
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+      ],
+    ]);
+
+    if ($response->getStatusCode() !== 200) {
+      throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+    }
+
+    try {
+      $token = json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR)['access_token'];
```

Comment:
This one liner seems prone to errors, for example, if the response doesn't have the access_token property.